### PR TITLE
👌 IMPROVE: CTRL-C on running process

### PR DESCRIPTION
**Pull request by chrisjsewell**
Do not call `kill` on a process that is already being killed,
Also log a different message,
so that the user can see that the original CTRL-C was actioned.

### CODE REVIEW

File: aiida/engine/runners.py [hunk 1]
```diff
@@ -235,6 +235,9 @@ def _run(self, process: TYPE_RUN_PROCESS, *args: Any, **inputs: Any) -> Tuple[Di
 
             def kill_process(_num, _frame):
                 """Send the kill signal to the process in the current scope."""
+                if process_inited.is_killing:
+                    LOGGER.critical('runner received interrupt, process %s already being killed', process_inited.pid)
```
Comment (ltalirz) [inline aiida/engine/runners.py:239]: does this need to be critical or would a warning suffice?
warning is anyhow the default log level

**Review by ltalirz — APPROVED** [review]
thanks @chrisjsewell !

that makes sense

File: aiida/engine/runners.py (hunk 1 above)
Comment (chrisjsewell) [inline aiida/engine/runners.py:239]:
```suggestion
                    LOGGER.warning('runner received interrupt, process %s already being killed', process_inited.pid)
```

Comment (chrisjsewell) [reply aiida/engine/runners.py:239]: yeh sounds good

**Pull request merged**